dev/devicelab/lib/framework/runner.dart

```diff
@@ -125,3 +127,33 @@ Future<VMIsolateRef> _connectToRunnerIsolate(int vmServicePort) async {
     }
   }
 }
+
+Future<void> cleanupSystem() async {
+  print('\n\nCleaning up system after task...');
+  final String javaHome = await findJavaHome();
+  if (javaHome != null) {
+    print('\nTelling Gradle to shut down (JAVA_HOME=$javaHome)');
+    final String gradlewBinaryName = Platform.isWindows ? 'gradlew.bat' : 'gradlew';
+    final Directory tempDir = Directory.systemTemp.createTempSync('flutter_devicelab_shutdown_gradle.');
+    recursiveCopy(new Directory(path.join(flutterDirectory.path, 'bin', 'cache', 'artifacts', 'gradle_wrapper')), tempDir);
+    copy(new File(path.join(path.join(flutterDirectory.path, 'packages', 'flutter_tools'), 'templates', 'create', 'android.tmpl', 'gradle', 'wrapper', 'gradle-wrapper.properties')), new Directory(path.join(tempDir.path, 'gradle', 'wrapper')));
+    if (!Platform.isWindows) {
+      await exec(
+        'chmod',
+        <String>['a+x', path.join(tempDir.path, gradlewBinaryName)],
+        canFail: true,
+      );
+    }
+    await exec(
+      path.join(tempDir.path, gradlewBinaryName),
+      <String>['--stop'],
+      environment: <String, String>{ 'JAVA_HOME': javaHome },
+      workingDirectory: tempDir.path,
+      canFail: true,
+    );
+    rmTree(tempDir);
```

Comment:
We've had issues in the past with Windows having mysterious file locks when we try to delete newly-created files in cleanup scripts, not sure if we may need to tolerate errors for this? (If you don't have easy access to Windows and want to check before landing, I can see if I can test this a little on my PC later)

Reply:
`rmTree` uses `rm` which tolerates errors.

Reply:
Doh! I was looking at old code, it looked like this:

```dart
/// Remove recursively.
void rmTree(FileSystemEntity entity) {
  if (entity.existsSync())
    entity.deleteSync(recursive: true);
}
```

But I see it's changed recently! :)